### ISSUE TITLE
Test suite broken on Firefox 89 - "Private fields are not currently supported"

### ISSUE THREAD

**Issue by Jaifroid** [jimmywarting/native-file-system-adapter#15]
Trying to run https://jimmywarting.github.io/native-file-system-adapter/example/test.html in Firefox 89 (latest stable), none of the tests work. In console I get an uncaught syntax error "Private fields are not currently supported" in `FileSystemDirectoryHandle.js:6:2`.

![imagen](https://user-images.githubusercontent.com/4304337/124745987-4da80c00-df18-11eb-9ec5-bfcfbfe69516.png)

**Comment by jimmywarting** [comment]
Hmm, guess i have to resort to using something older like WeakMap or Symbol.
for some reason i thought FF had already shipped private fields but not private methods.

Apparently it will be supported in next release (v90)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#browser_compatibility
Maybe just wait it out? a new FF release will happen soonish? and clients are good at auto update now...

Going to see if i can resolve it later today, @ work atm
Unless you would like to have a go at it?

would basically have to switch out
https://github.com/jimmywarting/native-file-system-adapter/blob/ac1693c2d5d7cc671d79526b38f1c87fd273a528/src/FileSystemDirectoryHandle.js#L4-L11

to something like 
```js
const _adapter = Symbol('adapter')
 class FileSystemDirectoryHandle extends FileSystemHandle { 
   /** @type {FileSystemDirectoryHandle} */ 
   [_adapter]
   
  constructor(adapter) {
     super(adapter)
     this[_adapter] = adapter
   }
```

or something even more private

```js
const wm = new WeakMap()

const adapter = Symbol('adapter')
 class FileSystemDirectoryHandle extends FileSystemHandle { 
   constructor(adapter) {
     super(adapter)
     wm.set(this, adapter)
   }
```

**Comment by Jaifroid** [comment]
Thanks for quick diagnosis. Do you promise specific browser support, e.g. FF >= 80 or whatever? I know you say you support browsers which implement [JavaScript modules](https://caniuse.com/?search=module), but you don't seem to specify actual browser versions. Just depends on the purpose of this ponyfilll, I guess! (I don't think I'm a good enough dev to contribute as I don't yet use this -- I'm just scouting around for polyfills for File Access that I can use in my project at this stage, and yours looks great!)

**Comment by jimmywarting** [comment]
I'm not running any automated test atm that test different browsers (would like to have it doe...)

the lowest thing i want to have is support for top-level await which means ESM, the rest can be downgraded.

**Comment by jimmywarting** [comment]
fixed in cbd8dcdf1a96b496a74a4218cff02934c703c771 
npm relase will happen soonish within minutes or so

the test works on the web page now

**Comment by Jaifroid** [comment]
I can confirm that the test page now works in Firefox 89. Many thanks for the quick fix!
